connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/memory/AgentContextChatMemoryStore.java

```diff
@@ -0,0 +1,103 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageDeserializer;
+import dev.langchain4j.data.message.ChatMessageSerializer;
+import dev.langchain4j.store.memory.chat.ChatMemoryStore;
+import dev.langchain4j.store.memory.chat.InMemoryChatMemoryStore;
+import io.camunda.connector.agenticai.aiagent.model.AgentContext;
+import io.camunda.connector.api.error.ConnectorException;
+import java.util.List;
+import java.util.Map;
+
+public class AgentContextChatMemoryStore implements ChatMemoryStore {
+
+  public static final String DEFAULT_MEMORY_ID = "default";
+
+  private final ChatMemoryStore internalMemoryStore;
+  private final ObjectMapper objectMapper;
+
+  public AgentContextChatMemoryStore(ObjectMapper objectMapper) {
+    this(new InMemoryChatMemoryStore(), objectMapper);
+  }
+
+  public AgentContextChatMemoryStore(
+      ChatMemoryStore internalMemoryStore, ObjectMapper objectMapper) {
+    this.internalMemoryStore = internalMemoryStore;
+    this.objectMapper = objectMapper;
+  }
+
+  @Override
+  public List<ChatMessage> getMessages(Object memoryId) {
+    return internalMemoryStore.getMessages(memoryId);
+  }
+
+  @Override
+  public void updateMessages(Object memoryId, List<ChatMessage> messages) {
+    internalMemoryStore.updateMessages(memoryId, messages);
+  }
+
+  @Override
+  public void deleteMessages(Object memoryId) {
+    internalMemoryStore.deleteMessages(memoryId);
+  }
+
+  public void loadFromAgentContext(AgentContext agentContext) {
+    loadFromAgentContext(agentContext, DEFAULT_MEMORY_ID);
+  }
+
+  public void loadFromAgentContext(AgentContext agentContext, Object memoryId) {
+    final var messages = agentContext.memory().stream().map(this::deserializeChatMessage).toList();
+
+    this.updateMessages(memoryId, messages);
+  }
+
+  public AgentContext storeToAgentContext(AgentContext agentContext) {
+    return storeToAgentContext(agentContext, DEFAULT_MEMORY_ID);
+  }
+
+  public AgentContext storeToAgentContext(AgentContext agentContext, Object memoryId) {
+    final var messages =
+        this.getMessages(memoryId).stream().map(this::serializeChatMessage).toList();
+
+    return agentContext.withMemory(messages);
+  }
+
+  private Map<String, Object> serializeChatMessage(ChatMessage chatMessage) {
+    // step 1: convert to JSON string from LangChain4J internal structure
+    String json = ChatMessageSerializer.messageToJson(chatMessage);
+
+    // step 2: convert JSON string to map for better variable inspection
+    try {
+      return objectMapper.readValue(json, STRING_OBJECT_MAP_TYPE_REFERENCE);
+    } catch (JsonProcessingException e) {
+      throw new ConnectorException(
+          "Failed to convert memory entry: " + humanReadableJsonProcessingExceptionMessage(e));
+    }
+  }
+
```

Comment:
We're using Langchain4j's serialization format for now, but that exposes only serializing to a JSON string which is not too nice to introspect in Operate. So we first serialize to string and then parse that JSON structure into a Map.

In the future we can improve this, but for now I would go this way for a first iteration. Examples how we could improve:

- using our own data structures for storage
- finding ways to serialize to a Map directly